tests/secretCRUD_test.go

```diff
@@ -0,0 +1,186 @@
+package tests
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"net/http"
+	"os"
+	"testing"
+
+	"github.com/openfaas/faas-provider/types"
+	"github.com/openfaas/faas/gateway/requests"
+)
+
+var secretsURL string = os.Getenv("gateway_url") + "system/secrets"
+var swarm = flag.Bool("swarm", false, "run swarm-compatible tests only")
+
+type secret types.Secret
+
+func Test_SecretCRUD(t *testing.T) {
+	setValue := "this-is-the-secret-value"
+	setName := "secret-name"
+	functionName := "test-secret-crud"
+
+	createStatus, err := createSecret(setName, setValue)
+	if err != nil {
+		t.Error(err)
+	}
+	if createStatus != http.StatusCreated && createStatus != http.StatusAccepted {
+		t.Errorf("got %d, wanted %d or %d", createStatus, http.StatusOK, http.StatusAccepted)
+	}
+	t.Logf("Got correct response for creating secret: %d", createStatus)
+
+	// Set up and deploy function that reads the value of the created secret.
+	functionRequest := requests.CreateFunctionRequest{
```

Comment:
If we are going to also test deploying a function, should we also have a test case that shows the function is rejected when the secret is missing?

Reply:
Sounds like a good idea, maybe in another PR after this one?

Reply:
I would prefer to put it in another PR. I tried to do 
`value = string(invoke(t, functionRequest.Service, "", http.StatusOK))` 
after the function was deleted, but it gave a 200 response. I'll need some time to figure out why I'm not getting the expected "Not Found" error from server. I can write the issue if @LucasRoesler thinks that's acceptable.

Reply:
Sure, a second PR sounds good

Reply:
The function will not immediately disappear, there is a margin for error and also the events take some time to process in Kubernetes - up to 5-10s for a Pod deletion. You'd need to "busy wait" for it to be removed from the list I guess.